### PR TITLE
PRO-1903 locale switcher refreshes its knowledge when opened

### PR DESCRIPTION
**Pull request by boutell**

### CODE REVIEW

File: modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBarLocale.vue [hunk 1]
```diff
@@ -96,22 +97,22 @@ export default {
...
-      );
-      this.localized = Object.fromEntries(
-        docs.results
-          .filter(doc => doc.aposLocale.endsWith(':draft'))
-          .map(doc => [ doc.aposLocale.split(':')[0], doc ])
-      );
-    }
-  },
   methods: {
+    async open() {
```
Comment (boutell) [inline modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBarLocale.vue:101]: Just moved this code.

File: modules/@apostrophecms/i18n/ui/apos/components/AposI18nLocalize.vue [hunk 2]
```diff
@@ -661,7 +661,7 @@ export default {
           }
         }
         // New documents only
-        newRelatedDocs = relatedDocs.filter(doc => unlocalizedIds.has(doc._id));
+        relatedDocs = relatedDocs.filter(doc => unlocalizedIds.has(doc._id));
```
Comment (boutell) [inline modules/@apostrophecms/i18n/ui/apos/components/AposI18nLocalize.vue:664]: Unrelated fix; this was producing console turds and breaking the filtering for the "new related documents only" case.

File: modules/@apostrophecms/ui/ui/apos/lib/i18next.js [hunk 3]
```diff
@@ -33,6 +33,10 @@ export default {
     // multiple layers of code, as a single `label`
     // property for instance
     Vue.prototype.$t = (phrase, options) => {
+      if (phrase == null) {
```
Comment (boutell) [inline modules/@apostrophecms/ui/ui/apos/lib/i18next.js:36]: Unrelated fix: don't log console turds if $t is passed null, as can happen if a button has no label, for instance.

Comment (gregvanbrug) [inline modules/@apostrophecms/ui/ui/apos/lib/i18next.js:36]: Should this check for strict equality? Maybe needs to check undefined and null?

**Review by gregvanbrug — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/ui/ui/apos/lib/i18next.js (hunk 3 above)
Comment (gregvanbrug) [reply modules/@apostrophecms/ui/ui/apos/lib/i18next.js:36]: Alright. I thought this was against our linter rules.

**Review by gregvanbrug — APPROVED** [review]

File: modules/@apostrophecms/ui/ui/apos/lib/i18next.js (hunk 3 above)
Comment (boutell) [reply modules/@apostrophecms/ui/ui/apos/lib/i18next.js:36]: It's a common practice for a really useful test (nullishness), so the standard linter rules permit this specific one. Not == in general.
